packages/common/src/hooks/useImageSize.ts

```diff
@@ -82,77 +97,121 @@ export const useImageSize = <
-  const getLargerImage = getNextImage<ImageSize, ImageSizes>((a, b) => a > b)
+  const getSmallerImage = useMemo(
+    () => getNextImage<ImageSize, ImageSizes>((a, b) => a < b),
+    []
+  )
+  const getLargerImage = useMemo(
+    () => getNextImage<ImageSize, ImageSizes>((a, b) => a > b),
+    []
+  )
 
-  const getImageSize = (): Maybe<URL> => {
+  const getImageSize = useCallback((): { url: Maybe<URL>; type: string } => {
     if (id === null || id === undefined || typeof id === 'string') {
-      return ''
+      return { url: '', type: 'empty' }
+    }
+
+    const fallbackImage = (url: URL) => {
+      setPreviousId(null)
+      return url
     }
 
     // No image sizes object
     if (!sizes) {
-      return fallbackImage('')
+      return { url: fallbackImage(''), type: 'none' }
     }
 
     // An override exists
     if (DefaultSizes.OVERRIDE in sizes) {
       const override: Maybe<URL> = sizes[DefaultSizes.OVERRIDE]
       if (override) {
-        return fallbackImage(override)
+        return { url: fallbackImage(override), type: 'override' }
       }
 
-      return defaultImage
+      return { url: defaultImage, type: 'default' }
     }
 
     // The desired size exists
     if (size in sizes) {
       const desired: Maybe<URL> = sizes[size]
       if (desired) {
-        return desired
+        return { url: desired, type: 'desired' }
       }
 
-      return defaultImage
+      return { url: defaultImage, type: 'default' }
     }
 
     // A larger size exists
     const larger: Maybe<URL> = getLargerImage(sizes, size)
     if (larger) {
-      return fallbackImage(larger)
-    }
-
-    // If no larger size exists, dispatch to get the desired size
-    // Don't dispatch twice for the same id
-    if (load && getPreviousId() !== id) {
-      setPreviousId(id)
-      // Request the desired size
-      dispatch(action(id, size))
+      return { url: fallbackImage(larger), type: 'larger' }
     }
 
     // A smaller size exists
     const smaller: Maybe<URL> = getSmallerImage(sizes, size)
     if (smaller) {
-      return fallbackImage(smaller)
+      return { url: fallbackImage(smaller), type: 'smaller' }
     }
 
-    return undefined
+    return { url: undefined, type: 'undefined' }
+  }, [
+    defaultImage,
+    getLargerImage,
+    getSmallerImage,
+    id,
+    size,
+    sizes,
+    setPreviousId
+  ])
+
+  let imageUrl: Maybe<URL>
+  let imageType: Maybe<string>
+
+  if (!onDemand) {
+    const { url, type } = getImageSize()
+    imageUrl = url
+    imageType = type
   }
 
-  // TODO: sk - disambiguate the return value so it can be typed
-  if (!onDemand) return getImageSize() as any
-  return getImageSize as any
+  const previousId = getPreviousId()
+
+  const handleFetchLargeImage = useCallback(
+    (imageType) => {
+      if (
+        load &&
+        !(id === null || id === undefined || typeof id === 'string') &&
+        previousId !== id &&
+        (imageType === 'smaller' || imageType === 'undefined')
```

Comment:
str undefined?

Reply:
oh i see nvm. pushing a change to change type to a string union

Reply:
Yeah that was v quick and dirty